### PR TITLE
Machine Learning Services - Enabling external script for package management only

### LINKED ISSUE #9059: ML Services Extension name inconsistent
Currently the ML Services extension has the name [`machineLearningServices`](https://github.com/microsoft/azuredatastudio/blob/master/extensions/machine-learning-services/package.json#L2)

The suggested format for extension names is lower-case with dashes (e.g. it would be `machine-learning-services`)

### PR DESCRIPTION
**Pull request by llali**
Removed the server config view from the dashboard. Enabling external script is only required for package management 
Also fixes #9059

image

### CODE REVIEW

File: extensions/machine-learning-services/package.nls.json [hunk 1]
```diff
@@ -5,8 +5,8 @@
 	"title.documents": "Documents",
 	"title.configurations": "Configurations",
 	"title.endpoints": "Endpoints",
-	"mls.command.managePackages": "Manage Packages in SQL Server",
-	"mls.command.manageLanguages": "Manage External Languages",
+	"mls.command.managePackages": "Manage packages in database",
+	"mls.command.manageLanguages": "Manage external languages",
 	"mls.command.predictModel": "Make prediction",
 	"mls.command.manageModels": "Manage models",
```
Comment (udeeshagautam) [inline extensions/machine-learning-services/package.nls.json]: little outside the scope... manage models or deploy models?

File: extensions/machine-learning-services/package.json [hunk 2]
```diff
@@ -1,5 +1,5 @@
 {
-  "name": "machineLearningServices",
+  "name": "machine-learning-services",
```
Comment (udeeshagautam) [inline extensions/machine-learning-services/package.json]: please check that you don't have to change anywhere else - in build or excluded ext. or so?

File: extensions/machine-learning-services/src/packageManagement/packageManagementService.ts [hunk 3]
```diff
@@ -9,8 +9,9 @@ import { QueryRunner } from '../common/queryRunner';
 import * as constants from '../common/constants';
 import { ApiWrapper } from '../common/apiWrapper';
 import * as utils from '../common/utils';
+import * as nbExtensionApis from '../typings/notebookServices';
 
-export class ServerConfigManager {
+export class PackageManagementService {
```
Comment (udeeshagautam) [inline extensions/machine-learning-services/src/packageManagement/packageManagementService.ts]: its not used outisde ML ext? I am worried if this namechange would impact anything else?

File: extensions/machine-learning-services/src/controllers/mainController.ts [hunk 4]
```diff
@@ -169,9 +165,9 @@ export default class MainController implements vscode.Disposable {
 	/**
 	 * Returns the server config manager instance
 	 */
-	public get serverConfigManager(): ServerConfigManager {
+	public get serverConfigManager(): PackageManagementService {
```
Comment (udeeshagautam) [inline extensions/machine-learning-services/src/controllers/mainController.ts]: why not get packagaeManagementService()?

File: extensions/machine-learning-services/src/packageManagement/packageManagementService.ts [hunk 5]
```diff
@@ -76,15 +77,41 @@ export class ServerConfigManager {
...
 	 */
-	public async updateExternalScriptConfig(connection: azdata.connection.ConnectionProfile, enable: boolean): Promise<boolean> {
-		await this._queryRunner.updateExternalScriptConfig(connection, enable);
+	public async enableExternalScriptConfig(connection: azdata.connection.ConnectionProfile): Promise<boolean> {
 		let current = await this._queryRunner.isMachineLearningServiceEnabled(connection);
-		if (current === enable) {
-			this._apiWrapper.showInfoMessage(enable ? constants.mlsEnabledMessage : constants.mlsDisabledMessage);
+
+		if (current) {
+			return current;
```
Comment (udeeshagautam) [inline extensions/machine-learning-services/src/packageManagement/packageManagementService.ts]: nit: not sure why its showing an extra link between 83-84 extra blank spaces?

File: extensions/machine-learning-services/src/packageManagement/packageManagementService.ts [hunk 6]
```diff
@@ -76,15 +77,41 @@ export class ServerConfigManager {
 	 * @param connection SQL Connection
 	 * @param enable if true external script will be enabled
 	 */
-	public async updateExternalScriptConfig(connection: azdata.connection.ConnectionProfile, enable: boolean): Promise<boolean> {
-		await this._queryRunner.updateExternalScriptConfig(connection, enable);
+	public async enableExternalScriptConfig(connection: azdata.connection.ConnectionProfile): Promise<boolean> {
```
Comment (udeeshagautam) [inline extensions/machine-learning-services/src/packageManagement/packageManagementService.ts]: it seems we were try to incorporate enable/disable both but now only enable - why is that?

File: extensions/machine-learning-services/src/test/packageManagement/packageManagementService.test.ts [hunk 7]
```diff
@@ -23,74 +23,64 @@ function createContext(): TestContext {
...
 
 	it('openInstallDocuments should open document in browser successfully', async function (): Promise<void> {
 		const context = createContext();
 		context.apiWrapper.setup(x => x.openExternal(TypeMoq.It.isAny())).returns(() => Promise.resolve(true));
-		let serverConfigManager = new ServerConfigManager(context.apiWrapper.object, context.queryRunner.object);
+		let serverConfigManager = new PackageManagementService(context.apiWrapper.object, context.queryRunner.object);
 		should.equal(await serverConfigManager.openInstallDocuments(), true);
 	});
 
 	it('isMachineLearningServiceEnabled should return true if external script is enabled', async function (): Promise<void> {
```
Comment (udeeshagautam) [inline extensions/machine-learning-services/src/test/packageManagement/packageManagementService.test.ts]: nit: test for if not installed

File: extensions/machine-learning-services/package.nls.json (hunk 1 above)
Comment (llali) [reply extensions/machine-learning-services/package.nls.json]: still trying t figure that out. manage model  includes deploy and predict 


---
In reply to: [393904856](https://github.com/microsoft/azuredatastudio/pull/9519#discussion_r393904856) [](ancestors = 393904856)

File: extensions/machine-learning-services/package.json (hunk 2 above)
Comment (llali) [reply extensions/machine-learning-services/package.json]: yeah good point. will chck

---
In reply to: [393905206](https://github.com/microsoft/azuredatastudio/pull/9519#discussion_r393905206) [](ancestors = 393905206)

File: extensions/machine-learning-services/src/test/packageManagement/packageManagementService.test.ts [hunk 8]
```diff
@@ -23,74 +23,64 @@ function createContext(): TestContext {
...
 		context.apiWrapper.setup(x => x.showErrorMessage(TypeMoq.It.isAny())).returns(() => Promise.resolve(''));
-		let serverConfigManager = new ServerConfigManager(context.apiWrapper.object, context.queryRunner.object);
-		let connection  = new azdata.connection.ConnectionProfile();
-		await serverConfigManager.updateExternalScriptConfig(connection, true);
+		context.apiWrapper.setup(x => x.showQuickPick(TypeMoq.It.isAny(), TypeMoq.It.isAny(), TypeMoq.It.isAny())).returns(() => Promise.resolve({
+			label: 'Yes'
+		}));
+		let serverConfigManager = new PackageManagementService(context.apiWrapper.object, context.queryRunner.object);
+		let connection = new azdata.connection.ConnectionProfile();
+		await serverConfigManager.enableExternalScriptConfig(connection);
```
Comment (udeeshagautam) [inline extensions/machine-learning-services/src/test/packageManagement/packageManagementService.test.ts]: going forward we would probably want integration tests to actually do this. please correct me if we already have or dont need that.

File: extensions/machine-learning-services/src/packageManagement/packageManagementService.ts (hunk 3 above)
Comment (llali) [reply extensions/machine-learning-services/src/packageManagement/packageManagementService.ts]: no it's not. well nothing should use module from inside extension directly. it's not exposed as service also

---
In reply to: [393906056](https://github.com/microsoft/azuredatastudio/pull/9519#discussion_r393906056) [](ancestors = 393906056)

File: extensions/machine-learning-services/src/packageManagement/packageManagementService.ts (hunk 6 above)
Comment (llali) [reply extensions/machine-learning-services/src/packageManagement/packageManagementService.ts]: yes, we only check if it's not enabled and give user option to enable it. Based on the recent design, enable/disable server config should not be here. we should add a dialog for all configs in admin extension maybe

---
In reply to: [393908973](https://github.com/microsoft/azuredatastudio/pull/9519#discussion_r393908973) [](ancestors = 393908973)

File: extensions/machine-learning-services/src/test/packageManagement/packageManagementService.test.ts (hunk 8 above)
Comment (llali) [reply extensions/machine-learning-services/src/test/packageManagement/packageManagementService.test.ts]: I have integration test for sqltoolservice and will use sqltoolsservice api in another PR so removing the tsql from here. I don't think more integration tests needed but maybe I'm wrong

---
In reply to: [393911023](https://github.com/microsoft/azuredatastudio/pull/9519#discussion_r393911023) [](ancestors = 393911023)

File: extensions/machine-learning-services/package.json (hunk 2 above)
Comment (llali) [reply extensions/machine-learning-services/package.json]: verified not used anywhere 

---
In reply to: [393910789](https://github.com/microsoft/azuredatastudio/pull/9519#discussion_r393910789) [](ancestors = 393910789,393905206)

File: extensions/machine-learning-services/src/test/packageManagement/packageManagementService.test.ts (hunk 7 above)
Comment (llali) [reply extensions/machine-learning-services/src/test/packageManagement/packageManagementService.test.ts]: this seems s=correct. not sure what you mean

---
In reply to: [393910111](https://github.com/microsoft/azuredatastudio/pull/9519#discussion_r393910111) [](ancestors = 393910111)

**Comment by llali** [comment]
Added the screenshot

---
In reply to: [376332473](https://github.com/microsoft/azuredatastudio/pull/9519#issuecomment-376332473) [](ancestors = 376332473)

File: extensions/machine-learning-services/src/controllers/mainController.ts (hunk 4 above)
Comment (udeeshagautam) [reply extensions/machine-learning-services/src/controllers/mainController.ts]: public get **serverConfigManager**(): ?

File: extensions/machine-learning-services/src/packageManagement/packageManagementService.ts (hunk 6 above)
Comment (udeeshagautam) [reply extensions/machine-learning-services/src/packageManagement/packageManagementService.ts]: why not disable? (for my understanding)

File: extensions/machine-learning-services/src/test/packageManagement/packageManagementService.test.ts (hunk 7 above)
Comment (udeeshagautam) [reply extensions/machine-learning-services/src/test/packageManagement/packageManagementService.test.ts]: oh I meant - we are testing these functions return yes if ext script is enables and R is installed - but no negative test cases for when its not enabled/installed. But leave it to u to add or not...

File: extensions/machine-learning-services/src/packageManagement/packageManagementService.ts (hunk 6 above)
Comment (llali) [reply extensions/machine-learning-services/src/packageManagement/packageManagementService.ts]: this is prompted when user wants to do package management in database  which mean it needs the config to be enabled. there's not reason for user to disable it here. but user will have the option to disable any config in server config dialog which we have to add.

---
In reply to: [395374432](https://github.com/microsoft/azuredatastudio/pull/9519#discussion_r395374432) [](ancestors = 395374432)

File: extensions/machine-learning-services/src/controllers/mainController.ts (hunk 4 above)
Comment (llali) [reply extensions/machine-learning-services/src/controllers/mainController.ts]: I renamed the module and also renamed the property name here

---
In reply to: [395374192](https://github.com/microsoft/azuredatastudio/pull/9519#discussion_r395374192) [](ancestors = 395374192)

File: extensions/machine-learning-services/src/test/packageManagement/packageManagementService.test.ts (hunk 7 above)
Comment (llali) [reply extensions/machine-learning-services/src/test/packageManagement/packageManagementService.test.ts]: ah got it. sure will add that. maybe in another PR which I'm adding all the tests

---
In reply to: [395375016](https://github.com/microsoft/azuredatastudio/pull/9519#discussion_r395375016) [](ancestors = 395375016)

**Pull request merged**